### PR TITLE
Extended visualisation.tscn with c# example

### PR DESCRIPTION
**Pull request by SimonasLetukas**
https://github.com/MatejSloboda/Dijkstra_map_for_Godot/issues/124
https://github.com/MatejSloboda/Dijkstra_map_for_Godot/issues/122

### CODE REVIEW

**Comment by astrale-sharp** [comment]
I don't think this should appear in the gdscript part:
```

export var use_c_sharp: bool = false

var gridmap 

func _ready():
	if (use_c_sharp):
               .....
```

**Comment by SimonasLetukas** [comment]
> I don't think this should appear in the gdscript part:
> 
> ```
> 
> export var use_c_sharp: bool = false
> 
> var gridmap 
> 
> func _ready():
> 	if (use_c_sharp):
>                .....
> ```

Do you think that having two duplicated visualization scenes would be better than this checkbox approach? Keeping in mind that the whole Godot project would need to be mono-based anyway.

File: README.md [hunk 1]
```diff
@@ -127,6 +129,29 @@ my_dijkstra_map.recalculate(origin_point, {"terrain_weights": {TERRAIN_ID_FOREST
...
+## C# Support
+
+A wrapper located in `addons/dijkstra-map/Dijkstra_map_library/DijkstraMap.cs` can be used to interface with the library. [Example use](#examples) can be seen in `addons/dijkstra-map/visualization demo/visualisation.tscn`. The benefits of this wrapper: 
+
+* First-class development experience (same as GDScript).
+
+    In GDScript you can do:
+    ```GDScript
+    dijkstramap = DijkstraMap.new()
+    dijkstramap.add_square_grid(bmp)
```
Comment (astrale-sharp) [inline README.md:142]: what's bmp here?
Apart from that, it looks good, will be merged soon

**Comment by astrale-sharp** [comment]
> 

No, you're right, let's use the checkbox approach !

**Comment by SimonasLetukas** [comment]
Thanks for the comments! I'll update the PR and then test all the remaining endpoints sometime this week and mark this as non-WIP

**Comment by SimonasLetukas** [comment]
There we go, I also went through the endpoints and tested them manually, seems to be working as expected. My only worry is if the new library version would work for non-mono Godot projects? In theory the .gdns file hasn't changed so it should work the same, however the demo project was changed to mono and I'm unsure of the impact here.

**Comment by astrale-sharp** [comment]
If you could test it with a non mono godot version that would be amazing, otherwise, it'll have to wait till i come back home, the 10th this month, thanks for all your work!

**Comment by SimonasLetukas** [comment]
Non-mono new Godot project worked when tested with:
image

I then also created a new mono Godot project to see if the C# wrapper can be picked up:
image

And they both work :)

**Comment by astrale-sharp** [comment]
Well, I'll go with you on that one and if we missed something, we'll see by people complaining in the issues :innocent: 

Thank you very much for your involvement/work!

**Pull request merged**